adagrams/game.py

```diff
@@ -1,11 +1,107 @@
+import random
+LETTER_POOL = {
+    'A' : 9,	
+    'B' : 2,	
+    'C' : 2,	
+    'D' : 4,	
+    'E' : 12,	
+    'F' : 2,	
+    'G' : 3,	
+    'H' : 2,	
+    'I' : 9,	
+    'J' : 1,	
+    'K' : 1,	
+    'L' : 4,	
+    'M' : 2,
+    'N' : 6,
+    'O' : 8,
+    'P' : 2,
+    'Q' : 1,
+    'R' : 6,
+    'S' : 4,
+    'T' : 6,
+    'U' : 4,
+    'V' : 2,
+    'W' : 2,
+    'X' : 1,
+    'Y' : 2,
+    'Z' : 1
+}
 def draw_letters():
-    pass
+    # Create a list of keys from the LETTER_POOL dictionary and set it to the new variable "letters"
+    letters = list(LETTER_POOL.keys())   
+    # Once all the letters are added to the letters list, the list is shuffled
+    random.shuffle(letters)
+    # First 10 elements in the list are returned
+    return letters[:10]
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # Could also use copy.deepcopy() to create deep copy of letter_bank, but slicing works for this 
+    letterbank_copy = letter_bank[:]
+    # Loop through all the letters in the word, .upper() returns a string where all characters are in upper case
+    for letter in word.upper():
+        # If letter is not in the word
+        if letter not in letterbank_copy:
+            return False
+        else:
+            # Removes the letter from the 'letters' list and continue looping
+            letterbank_copy.remove(letter)
+    # If all the letters in the word are present in the letter_bank list, return True
+    return True
 
 def score_word(word):
-    pass
+    # Set initial value of score to zero
+    score = 0
+    # Use the .upper() function to make word upper case
+    word = word.upper()
+    for letter in word:
```

Comment:
We can apply `upper` to word in the `for` loop instead of reassigning `word` in the line prior.

```python
    for letter in word.upper():
```